### ISSUE TITLE
__precompile__ issues

### ISSUE THREAD

**Issue by cstjean** [JuliaLang/IJulia.jl#428]
On OSX, Julia 0.4.5, and the current IJulia release, if I delete `/libs/v0.4/IJulia.ji` and start the notebook server and a notebook, no precompilation occurs (`IJulia.ji` doesn't appear). I can also `using IJulia` from within the notebook without triggering it. From the REPL, it does trigger. If I trigger precompilation of a module that uses IJulia from within a notebook, I'll get `Module IJulia uuid did not match cache file. This is likely because module IJulia does not support  precompilation but is imported by a module that does.` But if I do it from the REPL, it works fine.

I've read some of the issues posted (eg. [#13200](https://github.com/JuliaLang/julia/issues/13200)). I haven't witnessed any intermittency.

In other words, I can use my precompilable IJulia-using module from the REPL, but not from a notebook.

**Comment by stevengj** [comment]
You can't precompile a module when that module is already loaded (as it is in the notebook). There is a Julia issue for this, but I'm on my phone and can't search for it now.

**Comment by cstjean** [comment]
[Found it](https://github.com/JuliaLang/julia/issues/12508). It makes some sense, but when I start a notebook, `IJulia` is already in `Main`, how did it get there without triggering its own precompilation, given that it has a `__precompile__` directive?

If I use my IJulia-using module in the REPL, it will precompile itself and IJulia correctly, but then using it in a fresh notebook will yield the "does not not match cache file" error as it tries to recompile them.

This is not a significant issue for me so I will close it down in favor of the JuliaLang one.

**Comment by stevengj** [comment]
Normally IJulia is precompiled when Julia is launched; I'm not sure why that's not happening.
